### ISSUE TITLE
Proposal: Adapt workflow example to be based on real source model

### ISSUE THREAD

**Issue by tortmayr** [eclipse-glsp/glsp#1183]
Currently the workflow example is based on the `direct-gmodel` library. While gmodel only is nice 
for quick prototyping and PoC creation most of the diagram languages implemented by our adopters are 
based on an actual underlying source model (for instance json or and EMF).
Since the workflow example is gmodel-only we basically never full test the API capabilities in development and a lot of bugs
remain undiscovered until reported by  an adopter that uses a different source model
(e.g. #1168)
IMO it would be better in the longrun if we adapt the workflow example to be based on an JSON source model. This way we have more
confidence that new API and features work as intend for a majority of adopters.

We could also use this opportunity to introduce a second diagram language a long side the workflow example that is GModel only.
This could be a read-only language that is just used to read and render GModel source models.
This way we could tackle multiple issues at once.
- We ensure that the main API and features are tested against real source model
- We have a second diagram language to test multi-editor integration and potential issues (see #804)
- We can add additional gmodel-only examples to test render specifics and client side features (e.g. edge caps, routing etc.)
- In the longrun this also makes it easier to test all capabilities and features ent-to-end(#1154 )

**Comment by planger** [comment]
I agree, this would help identifying potential flaws in new features earlier, also with respect to upcoming end-to-end tests.
However, the read-only use case I feel is not as important to set up.
